client/signup/controller.js

```diff
@@ -305,6 +306,21 @@ export default {
 		const isManageSiteFlow =
 			! excludeFromManageSiteFlows && ! isAddNewSiteFlow && isReEnteringSignupViaBrowserBack;
 
+		// Hydrate the store with domains dependencies from session storage,
+		// only in the onboarding flow.
+		const domainsDependencies = getDomainsDependencies();
+		if (
+			domainsDependencies &&
+			isManageSiteFlow &&
+			flowName === 'onboarding' &&
+			stepName !== 'domains'
```

Comment:
@allilevine this works for the current flow but something to be aware of is that this assumes the `onboarding` flow will always have only the domains and plans step. From p58i-i5q-p2, when the new signup flow is implemented which will have the domains step later in the flow, and potentially new steps added like goals and themes, then this will no longer work.

Reply:
@niranjan-uma-shankar Thanks for raising that! This is definitely specific to the current `onboarding` flow. To me, it's worth fixing these pain points until we implement the new flow. 

Has the implementation work started? And do you know if it will re-use these signup components? I'm hoping not. 😅

Reply:
I agree, this is definitely an improvement. 

> Has the implementation work started? And do you know if it will re-use these signup components? I'm hoping not. 😅

I'm not aware if the implementation has started. There's an upcoming experiment to test the impact of Stepper pbxNRc-4bM-p2. I think it's likely that we'll soon sunset this legacy framework and move to Stepper.

Reply:
I just tested browser back from checkout page on a [Stepper signup flow](https://wordpress.com/setup/new-hosted-site) and it has two issues:
* doesn't take me to the previous step, I get a blank page.
* Duplicate site is created.

Reply:
Anyhow, sorry for the digression, please feel free to deploy this PR :)